eo-maven-plugin/src/main/java/org/eolang/maven/MjUnphi.java

```diff
@@ -65,36 +65,45 @@ public final class MjUnphi extends MjSafe {
     @Override
     public void exec() {
         final List<String> errors = new CopyOnWriteArrayList<>();
-        final Path home = this.unphiOutputDir.toPath();
+        final Path input = this.unphiInputDir.toPath();
+        final Path output = this.unphiOutputDir.toPath();
         final Iterable<Directive> metas = new Phi.Metas(this.unphiMetas);
         final long start = System.currentTimeMillis();
         final int count = new Threaded<>(
-            new Walk(this.unphiInputDir.toPath()),
+            new Walk(input),
             phi -> {
-                final Path relative = this.unphiInputDir.toPath().relativize(phi);
                 final Path xmir = Paths.get(
-                    relative.toString().replace(
+                    input.relativize(phi).toString().replace(
                         String.format(".%s", MjPhi.EXT),
                         String.format(".%s", MjAssemble.XMIR)
                     )
                 );
-                final XML result = new Phi(phi, metas).unphi();
-                new Saved(result.toString(), home.resolve(xmir)).value();
-                Logger.debug(
-                    this,
-                    "Parsed to xmir: %[file]s -> %[file]s",
-                    phi, this.unphiOutputDir.toPath().resolve(xmir)
-                );
-                final List<String> here = MjUnphi.errors(result);
-                if (!here.isEmpty()) {
-                    errors.add(
-                        Logger.format(
-                            "%[file]s:\n\t%s\n",
-                            xmir,
-                            String.join("\n\t", here)
-                        )
-                    );
-                }
+                final Path target = output.resolve(xmir);
+                new FpDefault(
+                    source -> {
+                        final XML result = new Phi(source, metas).unphi();
+                        Logger.debug(
```

Comment:
@vl4ds4m these two lines are not connected anyhow, syntax wise. However, I believe, they must stay in exactly this order. This is a classic case of temporal coupling: https://www.yegor256.com/2015/12/08/temporal-coupling-between-method-calls.html To solve this, try to make the second line (`Logger.debug`) be dependent on the first one (`result =`).

Reply:
@yegor256 please check once again